### PR TITLE
Add support for scenes

### PR DESCRIPTION
**Pull request by IvarMatstoms**
Implemented parts of /scene api.
Added:
- Post to AbstractSmartHomeHub (required by /scene/<id>/trigger)
- New class Scene with attributes scene_id, name and icon
- Scene.trigger() to activate a scene
- Hub.get_scenes()
- Hub.get_scene_by_id(scene_id)

### CODE REVIEW

File: src/dirigera/hub/hub.py [hunk 1]
```diff
@@ -88,6 +90,17 @@ def get(self, route: str) -> Dict[str, Any]:
...
+    def post(self, route: str, data: Dict[str, Any] = None) -> Dict[str, Any]:
+        response = requests.post(
+            f"{self.api_base_url}{route}",
+            headers=self.headers(),
+            json=data,
+            timeout=10,
+            verify=False,
+        )
+        response.raise_for_status()
+        return response.text
```
Comment (Leggin) [inline src/dirigera/hub/hub.py:102]:
```suggestion
        return response.json()
```
without this the return type is wrong

File: src/dirigera/hub/hub.py [hunk 2]
```diff
@@ -19,11 +21,11 @@
 
 class Hub(AbstractSmartHomeHub):
     def __init__(
-        self,
-        token: str,
-        ip_address: str,
-        port: str = "8443",
-        api_version: str = "v1",
+            self,
```
Comment (Leggin) [inline src/dirigera/hub/hub.py:24]: why change the indentation here?

**Review by Leggin — CHANGES_REQUESTED** [review]
Thank your for the PR, I have some smaller suggestions.
And also please add tests.

File: src/dirigera/hub/hub.py (hunk 2 above)
Comment (IvarMatstoms) [reply src/dirigera/hub/hub.py]: Accidentally auto formatted, reverted

File: src/dirigera/hub/hub.py (hunk 1 above)
Comment (IvarMatstoms) [reply src/dirigera/hub/hub.py]: The api endpoint /scenes/<id>/trigger returns a empty body so performing response.json() will raise an exception, I changed return type hint to Optional[Dict[str, Any]] and returns None if response body is empty.

I would propose making the same changes to Hub.patch (the return value does seem to be used but right now the type hint is Dict[str, Any] but the actual return value is response.text). Although I am not aware of any POST or PATCH endpoint that returns a body so the return value might as well be removed.

**Comment by IvarMatstoms** [comment]
Tests added, currently testing:
* dict_to_scene
* Scene.trigger

**Comment by Leggin** [comment]
Looks good thank you!

File: src/dirigera/hub/hub.py (hunk 1 above)
Comment (Leggin) [reply src/dirigera/hub/hub.py]: Good point I will change this

**Pull request merged**